app/scripts/migrations/084.ts

```diff
@@ -0,0 +1,59 @@
+import { cloneDeep } from 'lodash';
+import { isObject } from '@metamask/utils';
+
+export const version = 84;
+
+/**
+ * Ensure that each networkConfigurations object in state.NetworkController.networkConfigurations has an
+ * `id` property which matches the key pointing that object
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (!isObject(state.NetworkController)) {
+    return state;
+  }
+  const { NetworkController }: any = state;
```

Comment:
Nit: What purpose is the `: any` serving here? It seems that it's making the type less specific (it's changing it from `RuntimeObject` to `any`)

Reply:
Oh I see, this is disabling type checking. The object spread operator below requires us to check that each of the old network configuration entries is an object, but we skipped that check because type checking was disabled here when this was set to `any`.

Reply:
`any` removed in 8116d9fe46

Reply:
The `any` was removed, but the required type checks have not been added yet. The type errors are the reason CI is failing

Reply:
To start with, I'd recommend not adding explicit types to variable declarations if the type can be inferred instead. The error here is a result of the inferred type being incompatible with the type you've declared.

It is necessary to declare a type in cases where there is no way for it to be inferred correctly. For example, the `newNetworkConfigurations` variable below, which is assigned to an empty object initially. The inferred type would be `{}`, which isn't correct, so an explicit type was required there. But here the type can be inferred directly from the assigned value.

Putting that issue aside, if you drop both unnecessary type declarations here, then `networkConfigurations` will end up as type `RuntimeObject` (e.g. `Record<string, unknown>`). This ends up failing on this line:

```
    newNetworkConfigurations[networkConfigurationId] = {
      ...networkConfigurations[networkConfigurationId],
      id: networkConfigurationId,
    };
```
with the error "Spread types may only be created from object types".

That is, you're assuming that the value (typed as `unknown`) is an object.

This can be fixed by adding another type check inside the loop, ensuring that any pre-existing configuration used is also an object. That's what I meant here:

>The object spread operator below requires us to check that each of the old network configuration entries is an object

Reply:
okay, thank-you

I have attempted to make improvements in line with your above comment here affa96040e